pkg/cmd/roachtest/roachstress.sh

```diff
@@ -16,6 +16,66 @@ if [ -z "${BASH_VERSINFO}" ] || [ -z "${BASH_VERSINFO[0]}" ] || [ "${BASH_VERSIN
   exit 1
 fi
 
+function help {
+  cat <<EOF
+Stress run roachtest
+Usage:
+  $(basename ${0}) [flags] [-- roachtest flags]
+
+flags:
+  -t - test to run (overrides TEST)
+  -c - number of tests to run (overrides COUNT)
+  -l - run test locally on host (overrides LOCAL)
+  -s - y/n buildshort instead of build (exclude web ui)
+  -b - cr | test | all - remove artifacts to force rebuild
```

Comment:
Rather than send a PR, let me describe here what I thought we'd do:

`if [ -n "$(git status --porcelain)" ]` would force a rebuild (i.e. set `-b all`). Additionally, we will add `-dirty` to `abase`. That way, the rebuild will not remove any artifacts that came from a clean SHA, and you won't generate more and more gunk as more local tests are carried out. At the end of this, `cr | test | all` doesn't really make much of a difference so I'd suggest simplifying to make this just a bool. This is probably fine anyway, since the build system already caches a lot of the work.
The nice upshot is that folks that make local modifications and don't pass `-b` still get the right behavior, i.e. their changes are included.

Reply:
I'm curious about `git status --porcelain` check. I usually have a bunch of untracked files/directories as a result of the build e.g. pkg/ui/ccl/, pkg/sql/lex/keywords.go. They might be residues from branch switching, but nevertheless I always have them and they would always force rebuild in that case.
As for `cr | test | all` nonsense, I don't like that myself, but i needed something quick. Mind that rebuilding problem is more macos specific as it involves slow docker so just rebuilding everything blindly is not ideal.

Reply:
Good point on the docker stuff - it's super costly there especially since the build cache is always busted.

I'm fine compromising on the details, but I think it's important that if I touch some CRDB file then the script won't silently reuse my cached builds. The sane behavior should be the default.

Btw, I don't know how you build with untracked files in sql/lex. These should cause problems. My checkout is always clean, I see to it that way, and I have separate worktrees that I use for the older release branches, where such cruft most prominently comes up.

Perhaps we use `git status --porcelain --untracked-files=no`? That seems like a good compromise - it will detect if you're changing some existing file, but it won't be thrown off by the random files you mention above.

Reply:
Maybe i can just use $(git diff | shasum | cut -d ' ' -f 1) and a sha of changes on top of revision and cache results like dirty, and replace previous dirty in that case. Then we won't build up dirty builds with every change and still have predictable rebuild with local changes.

Reply:
I understand where you're getting at, but I think this is getting a little out of hand - let's revisit the other option, which is that we require a clean checkout (`git diff --untracked-files=no`)? That way the script will just force you to commit each of your changes which means `git` will give you a unique SHA for each unique build product. It seems better after all to do it that way, not that we're verging into computing ad-hoc SHAs of uncommitted changes :-)

Reply:
I think `git status --porcelain --untracked-files=no` will do the trick. And if you want local changes and not rebuild every time you run, you need to commit the change. I'll just print a warning reminding to commit then.

Reply:
👍🏽 sounds good, let me know when you're ready for another review.